flixel/system/frontEnds/SoundFrontEnd.hx

```diff
@@ -136,7 +136,11 @@ class SoundFrontEnd
 		if (EmbeddedSound != null)
 		{
 			sound.loadEmbedded(EmbeddedSound, Looped, AutoDestroy, OnComplete);
+			sound.onLoad = OnLoad();
 			loadHelper(sound, Volume, Group, AutoPlay);
+			// Call OnlLoad() because the sound already loaded
+			if (OnLoad != null && sound._sound != null)
+				OnLoad();
```

Comment:
Is it correct to use this here? What if somebody uses FlxSound without SountFrontEnd?

Reply:
I don't know if I follow your question. This is the only method where getting an embedded sound has an OnLoad argument. if they use `mySound.loadEmbedded` there's no use for an onLoad callback

Reply:
`FlxSound` always has an `onLoad` field you can set.

Reply:
onComplete and onLoad get nulled when loadStream or loadEmbedded are called (I just copied how onComplete works). so there's no actual way to get onLoad to trigger when using loadEmbedded. Except through `FlxG.sound.load`
maybe I should rethink this? someone could try to set sound.onLoad() before loadEmbedded.

I think the best option is simply to remove `sound.onLoad` and call the OnLoad callback passed in without storing it. I only added it because I thought new callbacks should behave like existing callbacks, but it's not really needed here. and  things are just simpler without it.

Reply:
Simple sounds good.